samples/SimpleDataSource/Tables/WordTable.cs

```diff
@@ -0,0 +1,138 @@
+            "Statistics for words",                               // The Table must have a description
+            "Words");                                             // A category is optional. It useful for grouping different types of tables in the viewer's UI.
+
+        //
+        // Declare columns here. You can do this using the ColumnConfiguration class. 
+        // It is possible to declaratively describe the table configuration as well. Please refer to our Advanced Topics Wiki page for more information.
+        //
+        // The Column metadata describes each column in the table. 
+        // Each column must have a unique GUID and a unique name. The GUID must be unique globally; the name only unique within the table.
+        //
+        // The UIHints provides some hints to a viewer on how to render the column. 
+        // In this sample, we are simply saying to allocate at least 80 units of width.
+        //
+
+        private static readonly ColumnConfiguration FileColumn = new ColumnConfiguration(
+            new ColumnMetadata(new Guid("{91C69594-4079-4478-B1A0-3FEC01641D8D}"), "File", "The file containing the word."),
+            new UIHints { Width = 80 });
+
+        private static readonly ColumnConfiguration WordColumn = new ColumnConfiguration(
+           new ColumnMetadata(new Guid("{A669FC83-BF61-4604-8BB2-44E66FCA7062}"), "Word", "The actual word."),
+           new UIHints { Width = 80 });
+
+        private static readonly ColumnConfiguration CharacterCountColumn = new ColumnConfiguration(
+           new ColumnMetadata(new Guid("{E3056A08-D44D-4CD6-8158-503BDAEF899C}"), "Character Count", "The number of characters in the word."),
+           new UIHints { Width = 80 });
+
+        private static readonly ColumnConfiguration TimeColumn = new ColumnConfiguration(
+         new ColumnMetadata(new Guid("{54B4A016-9F78-4BAE-A0AB-AFDFBF33C3F1}"), "Time", "The time when the word is written to the file."),
+         new UIHints { Width = 80 });
+
+        public override void Build(ITableBuilder tableBuilder)
+        {
+            //
+            // Implement your columns here. 
+            // Columns are implemented via Projections, which are simply functions that map a row index to a data point.
+            //
+
+            //
+            // Pre-process the data to get a list of <filename, time, word> tuples.
+            //
+
+            var allWords = new List<Tuple<string, Timestamp, string>>();
+            foreach(var kvp in this.Lines)
+            {
+                foreach (var tuple in kvp.Value.SelectMany(x => x.Item2.Split((char[])null, StringSplitOptions.RemoveEmptyEntries).Select(y => new Tuple<Timestamp, string>(x.Item1, y))))
+                {
+                    allWords.Add(Tuple.Create(kvp.Key, tuple.Item1, tuple.Item2));
+                }
+            }
+
+            //
+            // Use Projection.Index() to get a base projection from row index to a <filename, word> tuple.
```

Comment:
shouldn't this be <filename, time, word> tuple?

Reply:
yep, old comment.  Time as later added